docs/design/crd-proposal-phase1.md

```diff
@@ -0,0 +1,215 @@
+The operator may, or may not, combine one or more share into a single smbd
+instance.
+
+SmbShare CRs will refer to one or more SmbSecurityDomain resources. The
+SmbSecurityDomain refernce will define the main security properties of the
+smbd that hosts the share. If two SmbShares are defined with different
+SmbSecurityDomains they may not be combined under one smbd.
+
+The listings below are not meant to be entirely complete but they outline
+the genearal direction to make the operaterator a fully-fledged tool to
+create shares that can be consumed by a variety of clients for various use
+cases. Some sections are deliberately marked "TBD" (to be decided) as
+placeholders for more detail that will be needed even in Phase 1.
+
+## SmbSecurityDomain
+
+Spec Options:
+* `mode` - enumerated string "user" or "active-directory" - Indicates that this is an AD joined instance or based on a local mapping of users
+* `users` - mapping - A new subsection pertaining to "user" options - required when mode = "user"
+  * `secret` - string - The name of a secret containing user and group definitions that will be used to create the local users and groups for the service
+* `localName` - string - Optional string used to indicate how this instance will
+  be referred to in an Active Directory
+* `realm` - string - The dns domain name of an Active Directory domain
+* `domains` - list-of-maps - A new subsection pertaining to detailed "active-directory" configuration (optional)
+  * `backend` - enumerated string - the name of a (supported) ID mapping backend
+  * `domainName` - string - the name of the domain being configured
+  * TBD - options to configure the backend, ID range, etc.
+
+
+## SmbShare
+
+Spec Options:
+* `shareName` - string - Optional string giving the share a full SMB compliant name. If unset, the name will be derived by the operator.
+* `storage` - mapping - A new subsection to configure the storage backing the share.  * `pvc` - mapping - A new subsection used to configure PVC backing storage.
+    * `name` - string - The name of a PVC, managed externally to the operator, for the share
+    * `spec` - mapping, embedded pvc spec - An embedded PVC spec that will be used to dynamically create a backing PVC for the share; sharing the lifecycle of the PVC with the share
+  * TBD - Any other more custom storage backends if needed
+* `securityDomain`  - string - The name of the SmbSecurityDomain CR associated with this share
+* `grouping` - string - Optional string that if undefined will be generated by the operator. The grouping is used by the operator as a hint to combine shares under one smbd service, depending on other settings
+* `clustering` - mapping - A new subsection pertaining to clustering smbd instances
+  * `mode` - enumerated string "none" or "ctdb" - This option tells the operator if this share should be clustered or not
+  * `minSize` - int - Minimum number of smbd instances when clustering
```

Comment:
Why minimum and not an exact size?

Reply:
I wasn't sure if we'd want to have a minSize and a maxSize in the future. But perhaps that is premature. Would you prefer if I changed it to just "size" and we can deal with the future in the future?

Reply:
Note from earlier discussion: minSize was chosen to allow for more flexible grouping...